tests/unix/test_lwt_unix.cppo.ml

```diff
@@ -22,6 +22,46 @@
 open Test
 open Lwt.Infix
 
+let assert_fd_closed = "ASSERT_FD_CLOSED"
+let assert_fd_open   = "ASSERT_FD_OPEN"
+
+let test_cloexec assertion flags =
+  if Sys.win32 then Lwt.return true
+  else
+    Lwt_unix.openfile "/dev/zero" (Unix.O_RDONLY :: flags) 0o644 >>= fun fd ->
+    let fd_ = Lwt_unix.unix_file_descr fd in
+    match Lwt_unix.fork () with
+      | 0 ->
+          Unix.putenv assertion (string_of_int @@ Obj.magic fd_);
+          (* There's no portable way to obtain the executable name (which
+           * may even no longer exist at this point), but argv[0] fortunately
+           * has the right value when the tests are run with "make test". *)
+          Unix.execv Sys.argv.(0) [||]
+      | n ->
+          Lwt_unix.close fd >>= fun () ->
+          Lwt_unix.waitpid [] n >>= function
+            | _, Unix.WEXITED 0 -> Lwt.return_true
+            | _ -> Lwt.return_false
```

Comment:
It looks like this is failing the automated builds with warning 4, because it does not list the cases for the `Unix.process_status` explicitly, and we have lots of warnings enabled, and build with warnings as errors in CI. I think in this case, it's best to just list the two remaining possible cases explicitly, so we fail out in the extremely unlikely event `Unix.process_status` changes one day.